apps/paymaster-nft-mint/src/libraries/aa-sdk/createModularAccountClientWithPaymaster.ts

```diff
@@ -1,29 +1,61 @@
-import { SmartAccountSigner } from '@alchemy/aa-core'
+import {
+  SmartAccountSigner,
+  UserOperationRequest,
+  createSmartAccountClient,
+  deepHexlify,
+  resolveProperties,
+} from '@alchemy/aa-core'
 
-import { createMultiOwnerModularAccountClient } from '@alchemy/aa-accounts'
-import { sepolia } from 'viem/chains'
-import { superchainPaymasterMiddleware } from '@/libraries/aa-sdk/superchainPaymasterMiddleware'
+import { createMultiOwnerModularAccount } from '@alchemy/aa-accounts'
 import { Chain, http } from 'viem'
+import { createSuperchainPaymasterClient } from '@/libraries/aa-sdk/createSuperchainPaymasterClient'
 
-export const createModularAccountClientWithPaymaster = ({
+export const createModularAccountClientWithPaymaster = async ({
   signer,
+  rpcUrl,
   paymasterRpcUrl,
   bundlerRpcUrl,
   chain,
 }: {
   signer: SmartAccountSigner
+  rpcUrl: string
   paymasterRpcUrl: string
   bundlerRpcUrl: string
   chain: Chain
 }) => {
-  return createMultiOwnerModularAccountClient({
+  const superchainPaymasterClient = createSuperchainPaymasterClient({
+    paymasterRpcUrl,
+    chain,
+  })
+
+  const modularAccount = await createMultiOwnerModularAccount({
+    signer: signer,
+    chain: chain,
+    transport: http(rpcUrl),
+  })
+
+  return createSmartAccountClient({
```

Comment:
this is easier to grok than the separate `superchainPaymasterMiddleware` function we had previously